src/cc_debug.c

```diff
@@ -72,13 +73,17 @@ debug_stacktrace(int skip_count)
     }
 
     free(symbols);
+
+    /* when segfault happens, currently it will loop printing
+     * stack trace forever, add abort to stop this */
+    abort();    
 #endif
 }
 
 void
 debug_assert(const char *cond, const char *file, int line, int panic)
 {
-    log_stderr("assert '%s' failed @ (%s, %d)", cond, file, line);
+    log_stderr("[tid=%p] assert '%s' failed @ (%s, %d)", pthread_self(), cond, file, line); 
```

Comment:
POSIX no longer requires `pthread_t` to be an alias of `unsigned long`, so that type cast becomes OS dependent. See bottom line of [this page](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/types.h.html). I think the numeric ID is not that user-friendly anyway, but I think your goal is a good one.

Here's what I propose, we can create a thread local variable called `thread_name` using `pthread_key_create()`, and assign it a meaningful value at `pthread_create()` (e.g. `"server"`, `"worker_1"`, ...). We can then modify the `debug.c` module to always look up that key using `pthread_getspecific()`, and add that info to the log. This call returns NULL if key is out of range, so it's safe to use even the caller didn't assign the proper name as long as we check for that in the `debug` module. https://linux.die.net/man/3/pthread_setspecific

Reply:
I have updated as discussed, but do you think we should use `pthread_setname_np` and `pthread_getname_np`?

Reply:
I think it is better to use `pthread_setname_np`, adding a global pthread_key is not ideal in logging module is not ideal, because if the module which sets name of the thread does not load logging module, it will stop compiling because it cannot find the global pthread_key variable (such as in test). 
Let me know what you think.

Reply:
Makes sense. My knowledge of the `pthread*` functions is not complete, `pthread_[s|g]etname_np` are good choices.

Reply:
updated